aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -1492,6 +1517,152 @@ def get_openapi_json_schema(
+            List of middlewares to be used for the swagger route.
+        """
+        from aws_lambda_powertools.event_handler.openapi.models import Server
+
+        if not swagger_base_url:
+
+            @self.get("/swagger.js", include_in_schema=False)
+            def swagger_js():
+                body = Path.open(Path(__file__).parent / "openapi" / "swagger_ui" / "swagger-ui-bundle.min.js").read()
+                return Response(
+                    status_code=200,
+                    content_type="text/javascript",
+                    body=body,
+                    compress=True,
+                )
+
+            @self.get("/swagger.css", include_in_schema=False)
+            def swagger_css():
+                body = Path.open(Path(__file__).parent / "openapi" / "swagger_ui" / "swagger-ui.min.css").read()
+                return Response(
+                    status_code=200,
+                    content_type="text/css",
+                    body=body,
+                    compress=True,
+                )
+
+        @self.get(path, middlewares=middlewares, include_in_schema=False)
+        def swagger_handler():
+            base_path = self._get_base_path()
+
+            if swagger_base_url:
+                swagger_js = f"{swagger_base_url}/swagger-ui-bundle.min.js"
+                swagger_css = f"{swagger_base_url}/swagger-ui.min.css"
+            else:
+                swagger_js = f"{base_path}/swagger.js"
+                swagger_css = f"{base_path}/swagger.css"
+
+            openapi_servers = servers or [Server(url=base_path)]
+
+            spec = self.get_openapi_json_schema(
+                title=title,
+                version=version,
+                openapi_version=openapi_version,
+                summary=summary,
+                description=description,
+                tags=tags,
+                servers=openapi_servers,
+                terms_of_service=terms_of_service,
+                contact=contact,
+                license_info=license_info,
+            )
+
+            body = f"""
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <title>Swagger UI</title>
+    <link rel="stylesheet" type="text/css" href="{swagger_css}">
+</head>
+
+<body>
+    <div id="swagger-ui">
+        Loading...
+    </div>
+</body>
+
+<script src="{swagger_js}"></script>
+
+<script>
+  var swaggerUIOptions = {{
+    dom_id: "#swagger-ui",
+    docExpansion: "list",
+    deepLinking: true,
+    filter: true,
+    spec: JSON.parse(`
+        {spec}
+    `.trim()),
+    presets: [
+      SwaggerUIBundle.presets.apis,
+      SwaggerUIBundle.SwaggerUIStandalonePreset
+    ],
+    plugins: [
+      SwaggerUIBundle.plugins.DownloadUrl
+    ]
+  }}
+
+  var ui = SwaggerUIBundle(swaggerUIOptions)
+</script>
+</html>
+            """.strip()
+
+            return Response(
+                status_code=200,
+                content_type="text/html",
+                body=body,
+                compress=True,
```

Comment:
I think we need to add a parameter if the customer wants to compress or not. Forcing this we can have issues with net :: ERR_CONTENT_DECODING_FAILED 200 (OK) when running locally.